### ISSUE TITLE
Annotate a field as non editable in a Custom Entity

### ISSUE THREAD

**Issue by mcockrellsana** [cofoundry-cms/cofoundry#338]
Hey Joel,

Is there anyway to create a property on a Custom Entity Model and annotate it to mark it non editable from the admin UI?  I'd like to have a field that is stored on the custom entity, but is used for legacy information and should not be editable.  Any way to do this currently?

**Comment by HeyJoel** [comment]
There's no built-in attribute, but there is a "readonly" form field that you could take advantage of by building a custom attribute. This could will allow you to use a `[ReadOnly]` annotation:

```csharp
[AttributeUsage(AttributeTargets.Property)]
public class ReadOnlyAttribute : Attribute, IMetadataAttribute
{
    public void Process(DisplayMetadataProviderContext context)
    {
        var modelMetaData = context.DisplayMetadata;
        modelMetaData.TemplateHint = "Readonly";
    }
}
```

It will only display the raw value of the field without any formatting, but that might be enough for you.

If you think this is useful I can look at adding it to the next release.

**Comment by mcockrellsana** [comment]
Excellent, worked very well, that's what I needed.

I think this is useful in cases where a custom entity stores data that might be coming from an import and is legacy data that you don't want people to modify, but want visible for audit purposes.

**Comment by HeyJoel** [comment]
Ok cool, I'll keep this open and add it to the next release.

**Comment by HeyJoel** [comment]
Fixed, will be released in 0.9.

The new `[ReadOnly]` attribute works differently to the one posted above, in that rather than using the readonly UI control, it will instead display a non-editable version of the usual form control. This allows the display formatting of the expect control e.g. dates or custom entities rather than just a text rendering of the value.
